### PR TITLE
mimic: osd/OSDMap: Replace get_out_osds with get_out_existing_osds

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/39422

### CODE REVIEW

File: src/osd/OSDMap.h [hunk 1]
```diff
@@ -699,7 +699,7 @@ class OSDMap {
 
   void get_all_osds(set<int32_t>& ls) const;
   void get_up_osds(set<int32_t>& ls) const;
-  void get_out_osds(set<int32_t>& ls) const;
+  void get_out_existing_osds(std::set<int32_t>& ls) const;
```
Comment (pdvian) [inline src/osd/OSDMap.h]: @badone Should we keep uniform set<int32_t> instead of std::set<int32_t> ?

Comment (badone) [reply src/osd/OSDMap.h]: @pdvian Yes. I would keep it uniform even though the fully qualified form is more desirable. If you want to create a seperate PR to qualify the std::set references that would be cool. Check out the PR that fully qualified these in master and let me know if I can help.

**Comment by neha-ojha** [comment]
> LGTM once inconsistencies mentioned above are resolved.

looks like your comment hasn't been addressed, which is holding up this PR from moving to "needs-qa"?

**Comment by pdvian** [comment]
@badone Let me check if we have PR in master for above fully qualified form in OSDMap.  @neha-ojha Thanks for ping.

File: src/osd/OSDMap.h (hunk 1 above)
Comment (pdvian) [reply src/osd/OSDMap.h]: @badone the commit-id a192c1a525c3a4f0aa50e1ff282b028fbbb0b14c has above fully qualified form for these functions. The PR#27255 is quite big and need separate backport PR though we donot have tracker for 27255. Let me know your thoughts.

**Comment by pdvian** [comment]
Waiting for @badone inputs.

File: src/osd/OSDMap.h (hunk 1 above)
Comment (neha-ojha) [reply src/osd/OSDMap.h]: @badone is out, let's go ahead without the `std::` for this PR.

Comment (pdvian) [reply src/osd/OSDMap.h]: Sure. Let me do the changes accordingly.

**Comment by badone** [comment]
LGTM

**Comment by yuriw** [comment]
[wip-yuri-testing-2019-07-08-2007-mimic](https://trello.com/c/QAzBp49o/659-wip-yuri-testing-2019-07-08-2007-mimic)

**Pull request merged**